### ISSUE TITLE
(not set) on Firebase Analytics dashboard

### ISSUE THREAD

**Issue by fgarcia5** [dpa99c/cordova-plugin-firebasex#416]
In the Analytics dashboard, the value **"(not set)"** is displayed as a visited screen and I haven't found the bug that is causing it.

Has this happened to someone else? Would you be so kind as to help me?

Thank you very much in advance. To continue, the details:

**Current behavior:**

I am tracking the display of my screens in Analytics, as indicated in the documentation:
`window.FirebasePlugin.setScreenName(pageName);`
making sure "pageName" is always a string.

In the Analytics dashboard, the "screen_firebase" parameter of the "screen_view" event is often set to "(not set)". As you can see below:
![scree_view firebase_screen](https://user-images.githubusercontent.com/11658144/83299330-56ce9180-a1cc-11ea-8838-cf95cabc2ab3.PNG)
![image](https://user-images.githubusercontent.com/11658144/83301609-50daaf80-a1d0-11ea-8370-a2d39354ea56.png)

I have inspected my code, even at the native level (Java) but couldn't detect why this value "(not set)" reaches Analytics.

**Expected behavior:**
I would like the value "(not set)" not to be displayed on the dashboard as it could be confusing and also very messy.

**Environment information**

- Cordova CLI version 
	- 9.0.0 (cordova-lib@9.0.1)
- Cordova platform version
	- android 8.1.0
- Plugins & versions installed in project (including this plugin)

	- cordova-open-native-settings 1.5.2 "Native settings"
	- cordova-plugin-androidx 1.0.2 "cordova-plugin-androidx"
	- cordova-plugin-androidx-adapter 1.1.1 "cordova-plugin-androidx-adapter"
	- cordova-plugin-app-version 0.1.9 "AppVersion"
	- cordova-plugin-appavailability 0.4.2 "AppAvailability"
	- cordova-plugin-calendar 5.1.5 "Calendar"
	- cordova-plugin-device 2.0.3 "Device"
	- **cordova-plugin-firebasex 9.1.1 "Google Firebase Plugin"**
	- cordova-plugin-geolocation 4.0.2 "Geolocation"
	- cordova-plugin-inappbrowser 3.1.0 "InAppBrowser"
	- cordova-plugin-network-information 2.0.2 "Network Information"
	- cordova-plugin-splashscreen 5.0.3 "Splashscreen"
	- cordova-plugin-statusbar 2.4.3 "StatusBar"
	- cordova-plugin-vibration 3.1.1 "Vibration"
	- cordova-plugin-whitelist 1.3.4 "Whitelist"
	- cordova-plugin-x-socialsharing 5.6.0 "SocialSharing"
	- cordova.plugins.diagnostic 5.0.1 "Diagnostic"
	- es6-promise-plugin 4.2.2 "Promise"

**Other information:**
I have seen that different developers have had this bug with Analytics but I haven't been able to determine a possible solution. Here are some related links that might be useful:

- https://stackoverflow.com/questions/52871794/not-set-is-showing-on-firebase-analytics-dashboard
- https://stackoverflow.com/questions/47432570/firebase-screen-class-not-set
- https://stackoverflow.com/questions/48546883/firebase-analytics-not-tracking-the-name-of-my-app-screens-screen-name-not-set

Thank you very much again and sorry for my bad English.

**Comment by dpa99c** [comment]
TL;DR: the appearance of "(not set)" is a caveat of using the native Firebase Analytics SDK in a Cordova environment and is not something that can be "fixed".

The Firebase Analytics SDK (along with all the Firebase SDKs) are intended to be used in a purely native app so it makes assumptions about the how screen name relates to the native view lifecyle - [see here](https://firebase.google.com/docs/analytics/screenviews).

In the case of iOS, it assumes each of your "screens" will have its own native UIViewController.
However, in a Cordova app, you only have one native UIViewController which wraps the Cordova Webview in which your JS-based webapp runs.
So until such time as the Cordova environment has been initialized and your JS is able to set a screen name such as "Home", then as far as Firebase is concerned, your native UIViewController (i.e. the Cordova app container) has "not set" the current screen name.

Similarly, for Android the assumption is that your Android app is purely native and therefore Firebase SDK is reliant on the Android Activity lifecycle.
Therefore Firebase Analytics assumes you will set the screen name in a native Activity lifecycle method and of course the Cordova Android app container doesn't do this in the Activity which contains the Cordova WebView, so again it is "not set" until such time as you set it in the JS of you Cordova app via this plugin.

It's possible this plugin could enable you to set some default value at app startup other than "(not set)" such as "Cordova loading" or some such, but this is pointless as it means the same:
your Cordova app hasn't loaded yet.

Therefore I'm closing this on the basis that there's nothing meaningful that the plugin can do and also if you understand that "(not set)" simply means "your app is still loading", then you can ignore/filter these entries from your screen views in Firebase Analytics.
